research/gnn-survey/layers.py

```diff
@@ -117,3 +117,73 @@ def call(self, inputs):
     attn = self.dropout(attn)
     output = tf.matmul(attn, x)
     return output
+
+
+class SpGraphAttnLayer(tf.keras.layers.Layer):
```

Comment:
Usually, we want to avoid abbreviations. SparseGraphAttnLayer would be a better name.

Reply:
OK!